src/server/game.cpp

```diff
@@ -1071,6 +1071,35 @@ bool game::process_turn(simple_wml::document& data, const socket_ptr user)
 				}
 			}
 		} else if (command->child("surrender")) {
+			size_t sidei = 0;
+
+			for(auto s : sides_) {
+				if(s == user) {
+					break;
+				}
+				++sidei;
+			}
+
+			if(sidei < sides_.size()) {
+				simple_wml::document cfg;
+				std::string playername;
+				cfg.root().set_attr("side", std::to_string(sidei + 1).c_str());
+
+				if(owner_ == user) {
```

Comment:
Why is this check necessary? What's the difference to just unconditionally doing "playername = username(user);"?

Reply:
If you surrender someone needs to take control. This code determines the next side to take control (via transfer_side_control).